cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -436,152 +434,28 @@ func (impl *OSBuildJobImpl) Run(job worker.Job) error {
 			}))
 
-		case *target.KojiTargetOptions:
-			// Koji for some reason needs TLS renegotiation enabled.
-			// Clone the default http transport and enable renegotiation.
-			transport := http.DefaultTransport.(*http.Transport).Clone()
-			transport.TLSClientConfig = &tls.Config{
-				Renegotiation: tls.RenegotiateOnceAsClient,
-			}
-
-			kojiServer, _ := url.Parse(options.Server)
-			creds, exists := impl.KojiServers[kojiServer.Hostname()]
-			if !exists {
-				r = append(r, fmt.Errorf("Koji server has not been configured: %s", kojiServer.Hostname()))
-				continue
-			}
-
-			k, err := koji.NewFromGSSAPI(options.Server, &creds, transport)
-			if err != nil {
-				r = append(r, err)
-				continue
-			}
-
-			defer func() {
-				err := k.Logout()
-				if err != nil {
-					log.Printf("koji logout failed: %v", err)
-				}
-			}()
-
-			if !osbuildOutput.Success {
-				err = k.CGFailBuild(int(options.BuildID), options.Token)
-				if err != nil {
-					log.Printf("CGFailBuild failed: %v", err)
-				}
-				continue
-			}
-
-			f, err := os.Open(path.Join(outputDirectory, options.Filename))
-			if err != nil {
-				r = append(r, err)
-				continue
-			}
-
-			hash, filesize, err := k.Upload(f, options.UploadDirectory, options.KojiFilename)
-			if err != nil {
-				r = append(r, err)
-				continue
-			}
-
-			hostOS, err := distro.GetRedHatRelease()
-			if err != nil {
-				r = append(r, err)
-				continue
-			}
-
-			build := koji.ImageBuild{
-				BuildID:   options.BuildID,
-				TaskID:    options.TaskID,
-				Name:      options.Name,
-				Version:   options.Version,
-				Release:   options.Release,
-				StartTime: start_time.Unix(),
-				EndTime:   end_time.Unix(),
-			}
-			buildRoots := []koji.BuildRoot{
-				{
-					ID: 1,
-					Host: koji.Host{
-						Os:   hostOS,
-						Arch: common.CurrentArch(),
-					},
-					ContentGenerator: koji.ContentGenerator{
-						Name:    "osbuild",
-						Version: "0", // TODO: put the correct version here
-					},
-					Container: koji.Container{
-						Type: "none",
-						Arch: common.CurrentArch(),
-					},
-					Tools: []koji.Tool{},
-					RPMs:  osbuildStagesToRPMs(osbuildOutput.Build.Stages),
-				},
-			}
-			output := []koji.Image{
-				{
-					BuildRootID:  1,
-					Filename:     options.KojiFilename,
-					FileSize:     uint64(filesize),
-					Arch:         common.CurrentArch(),
-					ChecksumType: "md5",
-					MD5:          hash,
-					Type:         "image",
-					RPMs:         osbuildStagesToRPMs(osbuildOutput.Stages),
-					Extra: koji.ImageExtra{
-						Info: koji.ImageExtraInfo{
-							Arch: "noarch",
-						},
-					},
-				},
-			}
-
-			_, err = k.CGImport(build, buildRoots, output, options.UploadDirectory, options.Token)
-			if err != nil {
-				r = append(r, err)
-				continue
-			}
+			osbuildJobResult.Success = true
+			osbuildJobResult.UploadStatus = "success"
 		default:
-			r = append(r, fmt.Errorf("invalid target type"))
+			err = fmt.Errorf("invalid target type")
```

Comment:
You could optionally also log the target type here to have more information for debugging possible issues in the future. Not needed, since it was never done before.